src/fides/api/ctl/database/samples.py

```diff
@@ -0,0 +1,151 @@
+
+    Returns a dictionary, grouped by resource type (e.g. Dataset, System...)
+
+    NOTE: This only implements parsing for types we _currently_ expect to use in
+    the sample project. If new types are added to the project, this will quietly
+    ignore them unless strict=True is provided, which throws an error. We test
+    the "strict" version of this in the integration tests to ensure this code
+    doesn't fall out of date.
+    """
+    sample_resources_path = files(fides.data.sample_project).joinpath(
+        "sample_resources/"
+    )
+    sample_resources_dict: Dict[str, List] = {
+        "dataset": [],
+        "organization": [],
+        "policy": [],
+        "system": [],
+    }
+    for yaml_path in sample_resources_path.iterdir():
+        with yaml_path.open("r") as file:
+            # NOTE: in load_sample_connections, we use `expandvars` to expand
+            # ENV vars into string to allow injecting secrets, etc. We don't do
+            # that for the sample_resources/*, however, to stay consistent with
+            # the behaviour of `fides push`
+            yaml_dict = yaml.safe_load(file)
+
+            # Parse the contents of the YAML into the respective Pydantic models
+            if yaml_dict.get("dataset", []):
+                datasets = [Dataset.parse_obj(e) for e in yaml_dict.get("dataset")]
+                sample_resources_dict["dataset"].extend(datasets)
+            elif yaml_dict.get("organization", []):
+                organizations = [
+                    Organization.parse_obj(e) for e in yaml_dict.get("organization")
+                ]
+                sample_resources_dict["organization"].extend(organizations)
+            elif yaml_dict.get("policy", []):
+                policies = [Policy.parse_obj(e) for e in yaml_dict.get("policy")]
+                sample_resources_dict["policy"].extend(policies)
+            elif yaml_dict.get("system", []):
+                systems = [System.parse_obj(e) for e in yaml_dict.get("system")]
+                sample_resources_dict["system"].extend(systems)
+            else:
+                if strict:
+                    unknown_resource_keys = list(yaml_dict.keys())
+                    raise NotImplementedError(
+                        f"Unexpected sample resource type, keys={unknown_resource_keys}"
+                    )
+                # quietly ignore failures when strict=False, so we only fail during tests
+    return sample_resources_dict
+
+
+class SampleConnection(CreateConnectionConfigurationWithSecrets):
+    """
+    Schema for creating a sample connection to initialize the database, usable
+    for both database & SaaS connections. Extends the existing models with an
+    optional 'dataset' field.
+    """
+
+    dataset: Optional[FidesKey] = None
+    """
+    (Optional) Valid fides_key for an existing dataset that should be used
+    for this ConnectionConfig. If a dataset does not exist for this key,
+    creation will fail.
```

Comment:
Even with this docstring, it was still a little confusing why this was here until I traced through your creation of the connection configs -

Reply:
this looks good!